src/librustc_metadata/rmeta/decoder.rs

```diff
@@ -586,15 +578,24 @@ impl<'a, 'tcx> CrateMetadataRef<'a> {
         &self.raw_proc_macros.unwrap()[pos]
     }
 
-    fn item_name(&self, item_index: DefIndex) -> Symbol {
+    fn item_name(&self, item_index: DefIndex, sess: &Session) -> Ident {
         if !self.is_proc_macro(item_index) {
-            self.def_key(item_index)
+            let name = self
+                .def_key(item_index)
                 .disambiguated_data
                 .data
                 .get_opt_name()
-                .expect("no name in item_name")
+                .expect("no name in item_name");
+            let span = self
+                .root
+                .per_def
+                .ident_span
+                .get(self, item_index)
+                .map(|data| data.decode((self, sess)))
+                .unwrap_or(DUMMY_SP);
+            Ident::new(name, span)
         } else {
-            Symbol::intern(self.raw_proc_macro(item_index).name())
+            Ident::with_dummy_span(Symbol::intern(self.raw_proc_macro(item_index).name()))
```

Comment:
The `Span` could be from `def_span` perhaps?

Reply:
We don't currently encode that - we only encode the `DefIndex`s for proc macros, so that we can map them back to the proc macro harness.

Reply:
`def_span` -> `get_span`.
We do encode it, that's how [`Span::def_site`](https://doc.rust-lang.org/nightly/proc_macro/struct.Span.html#method.def_site) from proc macro API gets its result.
See how it's decoded in `fn load_proc_macro`.